roles/benchmarking_deploy_coco_dataset/files/entrypoint.sh

```diff
@@ -5,107 +5,19 @@ set -u
-EOF
-
-declare -A DIR_PREFIXES=(
-    ["test2017.zip"]="zips"
-    ["train2017.zip"]="zips"
-    ["val2017.zip"]="zips"
-    ["annotations_trainval2017.zip"]="annotations"
-    ["image_info_test2017.zip"]="annotations"
-)
-
-if [[ -z "$CERT_FILE" ]]; then
-    echo "INFO: no cert file provided, downloading from upstream."
-
-    DATASET_BASE_URL="http://images.cocodataset.org"
-    echo "Using the upstream base URL: $DATASET_BASE_URL"
-
-elif [[ ! -e "$CERT_FILE" ]]; then
-    echo "FATAL: cert file not found ($CERT_FILE)"
-    exit 1
-fi
-
 echo "Installing 'unzip' ..."
 dnf install unzip -y --quiet
 
-for dataset_filename in $(cat checksum.sha256sum | cut -d' ' -f3);
-do
-    if [[ -f "$dataset_filename".extracted ]]; then
-        echo "$dataset_filename already downloaded and extracted, skipping."
-        continue
-    fi
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip awscliv2.zip
+./aws/install
+/usr/local/bin/aws configure import --csv file://${CERT_FILE}
 
-    grep "  $dataset_filename" checksum.sha256sum | sed "s/$dataset_filename/-/" > stdin.sha256sum
-
-    if [[ -e "$dataset_filename" ]]; then
-        echo "INFO: Found an existing '$dataset_filename'. Checking its sha256sum ..."
-        ls -lh "$dataset_filename"
-        if ! cat "$dataset_filename" | sha256sum --check "stdin.sha256sum"; then
-            echo "INFO: '$dataset_filename' check sum is invalid. Delete it."
-            ls -lh "$dataset_filename"
-            rm "$dataset_filename"
-        else
-            echo "INFO: $dataset_filename already available."
-            continue
-        fi
-    fi
-
-    url="$DATASET_BASE_URL"
-    if [[ -z "$CERT_FILE" ]]; then
-        url="${url}/${DIR_PREFIXES[$dataset_filename]}"
-        cert=""
-    else
-        cert="--cert $CERT_FILE"
-    fi
-    url="${url}/$dataset_filename"
-
-    echo "Downloading $url ..."
-    touch "${dataset_filename}.url"
-
-    if ! time \
-         curl \
-         $cert \
-         --silent  --fail --show-error \
-         --retry 999999 \
-         --retry-max-time $(($DOWNLOAD_MAX_TIME_MIN * 60)) \
-         --continue-at - \
-         "${url}" \
-            | tee "$dataset_filename" \
-            | sha256sum --check stdin.sha256sum;
-    then
-        echo "FATAL: failed to download/verify $dataset_filename ..."
-        exit 1
-    fi
-done
-
-for dataset_filename in $(cat checksum.sha256sum | cut -d' ' -f3);
-do
-    if [[ -f "$dataset_filename".extracted ]]; then
-        echo "$dataset_filename already extracted."
-        continue
-    fi
-    echo "Extracting $dataset_filename ..."
-    unzip "$dataset_filename" >/dev/null
-
-    touch "${dataset_filename}.extracted"
-    echo "Deleting $dataset_filename ..."
-    rm -f "$dataset_filename"
-done
+/usr/local/bin/aws s3 cp s3://psapcocobucket/ /storage --profile=cocobucket --recursive
 
 echo "All done!"
```

Comment:
even with S3, we _need_ a way to validate the content of the storage directory, and not download twice if the content is valid.

maybe counting the number of file in each of the directories would be enough, assuming that S3 download cannot corrupt files. Think that `aws s3 cp` may be interrupted (Pod killed) in the middle of the download.

also, can you download the different files (or extracted directories) one by one? so that if we add more content in the bucket (eg, a preprocessed version of the files), we do not download it as well